File: NomadMapApp/composeApp/src/androidMain/kotlin/org/nomad/mapapp/data/model/Company.kt
package org.nomad.mapapp.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Company(
    val id: Int,
    val nif: String,
    val name: String,
    val mail: String,
    val phone: Int,
    val tags: String,
    val score: Float,
    val isProvider: Boolean,
    val isRetail: Boolean,
    val address: Address?,
    val ratingsCount: Int,
    val averageRating: Float
) {
    fun getPhoneAsString(): String = phone.toString()
    fun getTagsList(): List<String> = tags.split(",").map { it.trim() }.filter { it.isNotEmpty() }
    fun getDisplayScore(): Float = if (ratingsCount > 0) averageRating else 0f
}

File: RazorPage/RazorPage/Models/Company.cs
namespace RazorPage.Models
{
    public class Company
    {
        public int Id { get; set; }
        public string NIF { get; set; }
        public string Name { get; set; }
        public string Mail { get; set; }
        public int Phone { get; set; }
        public string Tags { get; set; }
        public float Score { get; set; }
        public string Retail { get; set; }
        public bool IsProvider { get; set; }
        public bool isRetail { get; set; }
        public Address Address { get; set; }
        public int RatingCounts { get; set; }
        public double AverageRating { get; set; }

    }
}
